metrics/smithy-otel-metrics/async.go

```diff
@@ -0,0 +1,62 @@
+package smithyotelmetrics
+
+import (
+	"context"
+
+	"github.com/aws/smithy-go/metrics"
+	otelmetric "go.opentelemetry.io/otel/metric"
+)
+
+type asyncInstrument struct {
+	otel otelmetric.Registration
+}
+
+var _ metrics.AsyncInstrument = (*asyncInstrument)(nil)
+
+func (i *asyncInstrument) Stop() {
+	i.otel.Unregister()
+}
+
+// int64Observer wraps an untyped, multi-instrument OTEL Observer to Observe()
+// against a single int64 instrument.
+type int64Observer struct {
+	observer   otelmetric.Observer
+	instrument otelmetric.Int64Observable
+}
+
+var _ metrics.Int64Observer = (*int64Observer)(nil)
+
+func (o *int64Observer) Observe(_ context.Context, v int64, opts ...metrics.RecordMetricOption) {
+	o.observer.ObserveInt64(o.instrument, v, withMetricProps(opts...))
+}
+
+// adaptInt64CB wraps an OTEL async instrument callback, binding it to a single
+// int64 instrument.
+func adaptInt64CB(io otelmetric.Int64Observable, cb metrics.Int64Callback) otelmetric.Callback {
+	return func(ctx context.Context, o otelmetric.Observer) error {
+		cb(ctx, &int64Observer{o, io})
```

Comment:
I may be blind, but I don't see where `cb` is defined

Reply:
<img width="801" alt="image" src="https://github.com/user-attachments/assets/6e89c688-d9a6-45a9-8a4e-8982c9d6ab49">

Reply:
Thanks for showing me my blindness :P